week5/hyungyu/src/FilterableProductTable.jsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react'
+import SearchBar from './SearchBar'
+import ProductTable from './ProductTable'
+import InputBar from './inputBar'
+
+const FilterableProductTable = () => {
+  const [product, setProduct] = useState([
+    {id: "1", category: "Fruits", price: 1, stocked: true, name: "Apple"},
+    {id: "2", category: "Fruits", price: 1, stocked: true, name: "Dragonfruit"},
+    {id: "3", category: "Fruits", price: 2, stocked: false, name: "Passionfruit"},
+    {id: "4", category: "Vegetables", price: 2, stocked: true, name: "Spinach"},
+    {id: "5", category: "Vegetables", price: 4, stocked: false, name: "Pumpkin"},
+    {id: "6", category: "Vegetables", price: 1, stocked: true, name: "Peas"}
+  ]);
+
+  const [filterText, setFilterText] = useState("");
+  const [inStockOnly, setInStockOnly] = useState(false);
+
+  const addProduct = (newProduct) => {
+    setProduct((previousData) => [...previousData, newProduct]);
+  };
+
+  const deleteProduct = (deletedProductId) => {
+    const deletedProducts = [];
+    for(let i = 0; i < product.length; i++){
+      if(product[i].id !== deletedProductId){
+        deletedProducts.push(product[i]);
+      }
+    }
+    setProduct(deletedProducts);
```

Comment:
setState 함수는 이전 상태(현재 저장된 state)를 콜백으로 불러올 수 있게 지원합니다.

이를 이용한다면 setProduct 안에서 더욱 간단히 표현 가능합니다!
```suggestion
    setProduct((prevData) => prevData.filter(  
      (item) => item.id !== deletedProductId  
    ));
```